pkgs/development/r-modules/default.nix

```diff
@@ -68,9 +68,10 @@ let
     hydraPlatforms = [];
   };
   deriveCran = mkDerive {
-    mkHomepage = {name, snapshot, ...}: "https://cran.r-project.org/${snapshot}/web/packages/${name}/";
-    mkUrls = {name, version, snapshot}: [
-      "https://packagemanager.rstudio.com/cran/${snapshot}/src/contrib/${name}_${version}.tar.gz"
+    mkHomepage = {name, ...}: "https://cran.r-project.org/web/packages/${name}/";
+    mkUrls = {name, version}: [
+      "mirror://cran/${name}_${version}.tar.gz"
+      "mirror://cran/Archive/${name}/${name}_${version}.tar.gz"
```

Comment:
Could we add native support for configurable mirrors? That would help me plug in my private and posit mirrors. The current default.nix buries deriveCran inside the let statement, which makes it awkward to patch or override. Some options:
1) We could add native support by exposing an argument
2) We could move deriveCran (and the other derive tools) outside the let so they're easy to override
3) We could write something like derivePrivate to extend the package set

Reply:
Support for private mirrors would be useful but maybe it's outside the scope of this PR?

Reply:
Sure, a separate PR if fine.

Reply:
Overall I agree the rPackages tree needs to be easier to override (not in this PR), but your particular use case is not impacted by this. You can edit `generate-r-packages.R` to pull the package index from posit for your selected date, and not touch anything else (i.e., continue pulling the actual package from CRAN). To be clear, you can change line 28 to the following:
```
pkgs <- as.data.table(available.packages("https://packagemanager.rstudio.com/cran/2021-12-01/src/contrib", filters=c("R_version", "OS_type", "duplicates")), method="libcurl")
```

I tested the above, found a bug, and fixed that so it works as expected now. Note though that I don't want to support this use as it breaks the dependence between the R version and the CRAN/BioC releases. Something more sophisticated is needed to move back in time (e.g., https://github.com/TyberiusPrime/nixR).

Finally we should keep the scope of this PR to fixing the posit hash problem so we don't block the next NixOS release.